src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -1034,7 +1023,9 @@ def update_reference_person_and_relationships(
 
         # Handle extreme edge cases where there would not be a value to map to.
         population.loc[
-            population["relation_to_household_head"].isnull(), "relation_to_household_head"
+            (population.index.isin(households_to_update_idx))
```

Comment:
I think you can replace lines 1025-1029 with a `.fillna("Other nonrelative")` on line 1022.